OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseUnitOnlyOnYoutubeFragment.java

```diff
@@ -31,16 +30,19 @@ public void onCreate(Bundle savedInstanceState) {
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
-        View v = inflater.inflate(R.layout.fragment_course_unit_only_on_youtube, container, false);
-        ((TextView) v.findViewById(R.id.only_youtube_available_message)).setText(R.string.assessment_only_on_youtube);
-        v.findViewById(R.id.view_on_youtube_button).setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                Intent i = new Intent(Intent.ACTION_VIEW);
-                i.setData(Uri.parse(((VideoBlockModel) unit).getData().encodedVideos.youtube.url));
-                startActivity(i);
-            }
+        final View view = inflater.inflate(R.layout.fragment_course_unit_only_on_youtube, container, false);
+        if (environment.getConfig().getYoutubeInAppPlayerConfig().isYoutubeEnabled()) {
+            ((TextView) view.findViewById(R.id.only_youtube_available_message)).setText(R.string.assessment_needed_updating_youtube);
+            view.findViewById(R.id.update_youtube_button).setVisibility(View.VISIBLE);
+            view.findViewById(R.id.update_youtube_button).setOnClickListener(v -> {
+                BrowserUtil.open(getActivity(), AppConstants.BROWSER_PLAYSTORE_YOUTUBE_URI);
```

Comment:
We are just passing some HTTP link that opens the play store app to update the YouTube if installed, otherwise, open the web browser.

IMO, We can just use `BrowserUtil.open` method to do that.